src/drv_oculus_rift/rift-hmd-radio.c

```diff
@@ -0,0 +1,295 @@
+		goto fail;
+
+	c->joy_x_range_min = nx_json_get (obj, "JoyXRangeMin")->int_value;
+	c->joy_x_range_max = nx_json_get (obj, "JoyXRangeMax")->int_value;
+	c->joy_x_dead_min = nx_json_get (obj, "JoyXDeadMin")->int_value;
+	c->joy_x_dead_max = nx_json_get (obj, "JoyXDeadMax")->int_value;
+	c->joy_y_range_min = nx_json_get (obj, "JoyYRangeMin")->int_value;
+	c->joy_y_range_max = nx_json_get (obj, "JoyYRangeMax")->int_value;
+	c->joy_y_dead_min = nx_json_get (obj, "JoyYDeadMin")->int_value;
+	c->joy_y_dead_max = nx_json_get (obj, "JoyYDeadMax")->int_value;
+
+	c->trigger_min_range = nx_json_get (obj, "TriggerMinRange")->int_value;
+	c->trigger_mid_range = nx_json_get (obj, "TriggerMidRange")->int_value;
+	c->trigger_max_range = nx_json_get (obj, "TriggerMaxRange")->int_value;
+
+	array = nx_json_get (obj, "GyroCalibration");
+	for (i = 0; i < 12; i++)
+		c->gyro_calibration[i] = nx_json_item (array, i)->dbl_value;
+
+	c->middle_min_range = nx_json_get (obj, "MiddleMinRange")->int_value;
+	c->middle_mid_range = nx_json_get (obj, "MiddleMidRange")->int_value;
+	c->middle_max_range = nx_json_get (obj, "MiddleMaxRange")->int_value;
+
+	c->middle_flipped = nx_json_get (obj, "MiddleFlipped")->int_value;
+
+	array = nx_json_get (obj, "AccCalibration");
+	for (i = 0; i < 12; i++)
+		c->acc_calibration[i] = nx_json_item (array, i)->dbl_value;
+
+	array = nx_json_get (obj, "CapSenseMin");
+	for (i = 0; i < 8; i++)
+		c->cap_sense_min[i] = nx_json_item (array, i)->int_value;
+
+	array = nx_json_get (obj, "CapSenseTouch");
+	for (i = 0; i < 8; i++)
+		c->cap_sense_touch[i] = nx_json_item (array, i)->int_value;
+
+	/* LED model points */
+#if 0
+	const nx_json *led_model = nx_json_get (obj, "ModelPoints");
+	if (led_model->type != NX_JSON_OBJECT)
+		goto fail;
+
+	rift_led_init (&touch->leds, led_model->length);
+
+	for (i = 0; i < touch->led.num_points; i++) {
+		char name[8];
+		int j;
+
+		snprintf(name, 8, "Point%d", i);
+		array = nx_json_get (led_model, name);
+
+		double point[6];
+
+		for (j = 0; j < 6; j++)
+			point[j] = nx_json_item (array, j);
+
+		touch->led.points[i].x = point[0];
+		touch->led.points[i].y = point[1];
+		touch->led.points[i].z = point[2];
+		touch->led.normals[i].x = point[3];
+		touch->led.normals[i].y = point[4];
+		touch->led.normals[i].z = point[5];
+	}
+#endif
+
+  nx_json_free (nxj);
+	return 0;
+fail:
+	LOGW ("Unrecognised Touch Controller JSON data version %d\n%s\n", version_number, json);
+  nx_json_free (nxj);
+	return -1;
+}
+
+int rift_touch_get_calibration(hid_device *handle,
+		int device_id,
+		rift_touch_calibration *calibration)
+{
+	uint8_t hash[16];
+	uint16_t length;
+	char *json = NULL;
+	int ret = -1;
+
+	/* If the controller isn't on yet, we might fail to read the calibration data */
+	ret = rift_radio_read_calibration_hash(handle, device_id, hash);
+	if (ret < 0) {
+		LOGV ("Failed to read calibration hash from device %d", device_id);
+		return ret;
+	}
+
+#if 0
```

Comment:
Same as above.

Reply:
This one is debug code. There's maybe scope for some general debugging functions like this in openhmd, that only get built when requested?